velox/exec/fuzzer/ArbitrationFuzzer.cpp

```diff
@@ -0,0 +1,400 @@
+  return builder.project({"c0", "c1", "c2"})
+      .orderBy({"c2 ASC NULLS LAST"}, false)
+      .planNode();
+}
+
+core::PlanNodePtr topNPlan(
+    const std::function<void(PlanBuilder&)>& makeSource) {
+  PlanBuilder builder;
+  makeSource(builder);
+  return builder.project({"c1"}).topN({"c1 NULLS FIRST"}, 10, false).planNode();
+}
+
+core::PlanNodePtr hashJoinPlan(const std::vector<RowVectorPtr>& inputs) {
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  return PlanBuilder(planNodeIdGenerator)
+      .values(inputs, true)
+      .project({"c0", "c1", "c2"})
+      .hashJoin(
+          {"c0"},
+          {"u0"},
+          PlanBuilder(planNodeIdGenerator)
+              .values(inputs, true)
+              .project({"c0 AS u0", "c1 AS u1", "c2 AS u2"})
+              .planNode(),
+          "",
+          {"c0", "c1", "c2"},
+          core::JoinType::kInner)
+      .planNode();
+}
+
+core::PlanNodePtr writePlan(
+    const std::function<void(PlanBuilder&)>& makeSource,
+    const std::string& outputDirPath) {
+  PlanBuilder builder;
+  makeSource(builder);
+  return builder.tableWrite(outputDirPath)
+      .singleAggregation(
+          {}, {fmt::format("sum({})", TableWriteTraits::rowCountColumnName())})
+      .planNode();
+}
+
+std::unordered_map<std::string, std::tuple<core::PlanNodePtr, RowVectorPtr>>
+ArbitrationFuzzer::makeDefaultPlanResults(
+    const std::vector<RowVectorPtr>& inputs) {
+  std::unordered_map<std::string, std::tuple<core::PlanNodePtr, RowVectorPtr>>
+      planResults;
+  for (const auto& name : operators) {
+    core::PlanNodePtr plan;
+    if (name == "OrderBy") {
+      plan = orderByPlan([&](PlanBuilder& builder) { builder.values(inputs); });
+    } else if (name == "HashJoin") {
+      plan = hashJoinPlan(inputs);
+    } else if (name == "TableWrite") {
+      const auto outputDirectory = TempDirectoryPath::create();
+      plan = writePlan(
+          [&](PlanBuilder& builder) { builder.values(inputs); },
+          outputDirectory->getPath());
+    } else if (name == "TopN") {
+      plan = topNPlan([&](PlanBuilder& builder) { builder.values(inputs); });
+    }
+    auto expected = AssertQueryBuilder(plan).copyResults(pool_.get());
+    planResults[name] = std::make_tuple(std::move(plan), std::move(expected));
+  }
+
+  return planResults;
+}
+
+void ArbitrationFuzzer::verify() {
+  // make default plans with input and expected.
+  auto stopGuard = folly::makeGuard([&]() { waitForAllTasksToBeDeleted(); });
+  const auto& inputs = generateInput();
+  const auto& planResults = makeDefaultPlanResults(inputs);
+  constexpr int numThreads = 32;
+  std::vector<std::thread> queryThreads;
+  queryThreads.reserve(numThreads);
+  int a = 0;
+  int b = 0;
+  const auto outputDirectory = TempDirectoryPath::create();
+  const auto spillDirectory = exec::test::TempDirectoryPath::create();
+  for (int i = 0; i < numThreads; ++i) {
+    queryThreads.emplace_back([&]() {
```

Comment:
We probably can skip the data verification for memory arbitration fuzzer test? Thanks!